packages/devtools_app/lib/src/memory/flutter/memory_heap_tree_view.dart

```diff
@@ -42,6 +42,73 @@ enum SnapshotStatus {
   done,
 }
 
+enum WildcardMatch {
+  exact,
+  startsWith,
+  endsWith,
+  contains,
+}
+
+/// If no wildcard then exact match.
+/// *NNN - ends with NNN
+/// NNN* - starts with NNN
+/// NNN*ZZZ - starts with NNN and ends with ZZZ
+const knowClassesToAnalyzeForImages = <WildcardMatch, List<String>>{
+  // Anything that contains the phrase:
+  WildcardMatch.contains: [
+    'Image',
+  ],
+
+  // Anything that starts with:
+  WildcardMatch.startsWith: [],
+
+  // Anything that exactly matches:
+  WildcardMatch.exact: [
+    '_Int32List',
+    'FrameInfos',
+  ],
+
+  // Anything that ends with:
+  WildcardMatch.endsWith: [],
+};
+
+/// RegEx expressions to handle the WildcardMatches:
+///     Ends with Image:      \[_A-Za-z0-9_]*Image\$
+///     Starts with Image:    ^Image
+///     Contains Image:       Image
+///     Extact Image:         ^Image$
+String buildRegExs(Map<WildcardMatch, List<String>> matchingCriteria) {
+  final resultRegEx = StringBuffer();
+  matchingCriteria.forEach((key, value) {
+    if (value.isNotEmpty) {
+      final name = value;
+      String regEx;
+      switch (key) {
+        case WildcardMatch.exact:
+          regEx = '^${name.join("\$|^")}\$';
```

Comment:
Add a TODO that this approach will blow up if the class names contain chars that impact regexes. E.g. searching for the class "$FOO" will cause problems.

Reply:
Done, good catch (I never use $ to start an identifier).